pyvista/plotting/lights.py

```diff
@@ -0,0 +1,466 @@
+
+    @property
+    def is_camera_light(self):
+        """Return whether the light is a camera light."""
+        return bool(self.LightTypeIsCameraLight())
+
+    @property
+    def is_scene_light(self):
+        """Return whether the light is a scene light."""
+        return bool(self.LightTypeIsSceneLight())
+
+    @property
+    def shadow_attenuation(self):
+        """Return the shadow intensity.
+
+        By default a light will be completely blocked when in shadow.
+        By setting this value to less than 1 you can control how much
+        light is attenuated when in shadow.
+
+        """
+        return self.GetShadowAttenuation()
+
+    @shadow_attenuation.setter
+    def shadow_attenuation(self, shadow_intensity):
+        """Set the shadow intensity."""
+        self.SetShadowAttenuation(shadow_intensity)
+
+    #### Everything else ####
+
+    def set_color(self, color):
+        # TODO: do we want to remove this?
+        # TODO: do we want instead to set _all_ lights in pyvista?
+        # TODO: do we want to support a three-arg form?
+        """Set the diffuse and specular color of the light.
+
+        Parameters
+        ----------
+        color : tuple(float)
+            The color that should be set for diffuse and specular.
+
+        """
+        self.SetColor(color)
+
+    def switch_on(self):
+        """Switch on the light."""
+        self.SwitchOn()
+
+    def switch_off(self):
+        """Switch off the light."""
+        self.SwitchOff()
+
+    def switch(self, is_on=None):
+        """Set whether the light should be on. Toggles by default.
+
+        Parameters
+        ----------
+        is_on : bool
+            The state of the light to be set (True is on, False is off).
+            Default is to toggle the state.
+
+        """
+        if is_on is None:
+            # toggle
+            is_on = not self.is_on
+        self.SetSwitch(is_on)
+
+    def positional_on(self):
+        """Make the light positional.
+
+        Attenuation and cone angles are only used for a positional light.
+
+        """
+        self.PositionalOn()
+
+    def positional_off(self):
+        """Make the light directional.
+
+        Attenuation and cone angles are ignored for a directional light.
+
+        """
+        self.PositionalOff()
+
+    # TODO: implement transform_point, transform_vector here?
+
+    def set_direction_angle(self, elev, azim):
+        """Set the position and focal point of a directional light.
+
+        The light is switched into directional (non-positional). The
+        position and focal point can be defined in terms of an elevation
+        and an azimuthal angle, both in degrees.
+
+        Parameters
+        ----------
+        elev : float
+            The elevation of the directional light.
+
+        azim : float
+            The azimuthal angle of the directional light.
+
+        """
+        self.SetDirectionAngle(elev, azim)
+
+    # TODO: deepcopy?
```

Comment:
Or just ``copy`` with a ``deep`` bool parameter as in our other classes.

Reply:
I have a technical issue with that. I suspect shallow copies should essentially implement [`vtkLight::ShallowClone()`](https://vtk.org/doc/nightly/html/classvtkLight.html#ac7bef205f8a78c3c9081db025e0892f2), but I don't understand vtk enough to know what this would entail for the pyvista class. At for deepcopy I assume it should return a new copy (as per python customs) rather than being a void method on a light object as [`vtkLight::DeepCopy()`](https://vtk.org/doc/nightly/html/classvtkLight.html#abfaf6d69238978415d61841ec6c2ff43). This is why I only poked the deepcopy possibility :)

Reply:
OK, I was being confused again. I've got this one working too.